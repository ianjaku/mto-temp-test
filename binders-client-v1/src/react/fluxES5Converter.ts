/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
