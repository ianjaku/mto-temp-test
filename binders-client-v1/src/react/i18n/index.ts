import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
