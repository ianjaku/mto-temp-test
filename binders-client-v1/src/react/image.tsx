/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
