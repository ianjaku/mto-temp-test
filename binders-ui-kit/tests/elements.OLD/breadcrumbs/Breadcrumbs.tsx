import * as React from "react";
import Breadcrumbs from "../../../src/elements/breadcrumbs";
import { MemoryRouter } from "react-router";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

let exampleValue = 100;
const onClick = () => exampleValue++;

const items = [
    {
        link: "/",
        name: "Home",
    },
    {
        link: "/lib",
        name: "My Library",
    },
    {
        link: "/colA",
        name: "Collection A",
    },
    {
        link: "/manB",
        name: "Manual B",
    },
];


const wrapWithThemeAndRouter = (crumbs) => (
    wrapWithTheme(<MemoryRouter>{crumbs}</MemoryRouter>)
);

const breadcrumbsWithOnClicks = () =>
    wrapWithThemeAndRouter(<Breadcrumbs items={items} onClick={onClick} />);
const normalBreadcrumbs = () => wrapWithThemeAndRouter(<Breadcrumbs items={items} />);
const singleBreadcrumb = () => wrapWithThemeAndRouter(<Breadcrumbs items={[items[0]]} />);

describe("Breadcrumbs", () => {
    beforeEach(() => {
        exampleValue = 100;
    });
    test("normal breadcrumbs (snapshot)", () => {
        const modal = create(normalBreadcrumbs());
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("single breadcrumb (snapshot)", () => {
        const modal = create(singleBreadcrumb());
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("breadcrumbs with onclicks (snapshot)", () => {
        const modal = create(breadcrumbsWithOnClicks());
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Breadcrumb single (mount: enzyme+jsdom)", () => {
        const element = mount(singleBreadcrumb()).find(".breadcrumbs-item");
        expect(element.length).toEqual(1);
    });

    test("Breadcrumb 4 items (mount: enzyme+jsdom)", () => {
        const element = mount(normalBreadcrumbs()).find("button");
        expect(element.length).toEqual(3);
    });

    test("Last one should be the active one", () => {
        const element = mount(normalBreadcrumbs()).find(".breadcrumbs-item");
        expect(element.at(9).hasClass("breadcrumbs-item--active")).toEqual(true);
    });

    test("Breadcrumbs with onClicks (shallow: enzyme+jsdom)", () => {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const button = mount(breadcrumbsWithOnClicks())
            .find(".breadcrumbs-item-link")
            .at(0)
            .simulate("click");
        expect(exampleValue).toEqual(101);
    });

    test("Breadcrumbs without onClicks (shallow: enzyme+jsdom)", () => {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const button = mount(normalBreadcrumbs())
            .find(".breadcrumbs-item-link")
            .at(0)
            .simulate("click");
        expect(exampleValue).toEqual(100);
    });
});
