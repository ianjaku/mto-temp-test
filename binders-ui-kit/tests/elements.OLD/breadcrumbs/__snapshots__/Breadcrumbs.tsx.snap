// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs breadcrumbs with onclicks (snapshot) 1`] = `
<div
  className="breadcrumbs"
>
  <a
    className="breadcrumbs-item"
    href="/"
    onClick={[Function]}
  >
    <button
      className="breadcrumbs-item-link"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": "auto",
          "lineHeight": "auto",
          "margin": 0,
          "minWidth": "50px",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        Home
      </div>
    </button>
  </a>
  <span
    className="material-icons breadcrumbs-arrow"
    color=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontSize": 24,
        "muiPrepared": true,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
      }
    }
  >
    keyboard_arrow_right
  </span>
  <a
    className="breadcrumbs-item"
    href="/lib"
    onClick={[Function]}
  >
    <button
      className="breadcrumbs-item-link"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": "auto",
          "lineHeight": "auto",
          "margin": 0,
          "minWidth": "50px",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        My Library
      </div>
    </button>
  </a>
  <span
    className="material-icons breadcrumbs-arrow"
    color=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontSize": 24,
        "muiPrepared": true,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
      }
    }
  >
    keyboard_arrow_right
  </span>
  <a
    className="breadcrumbs-item"
    href="/colA"
    onClick={[Function]}
  >
    <button
      className="breadcrumbs-item-link"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": "auto",
          "lineHeight": "auto",
          "margin": 0,
          "minWidth": "50px",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        Collection A
      </div>
    </button>
  </a>
  <span
    className="material-icons breadcrumbs-arrow"
    color=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontSize": 24,
        "muiPrepared": true,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
      }
    }
  >
    keyboard_arrow_right
  </span>
  <span
    className="breadcrumbs-item breadcrumbs-item--active"
  >
    Manual B
  </span>
</div>
`;

exports[`Breadcrumbs normal breadcrumbs (snapshot) 1`] = `
<div
  className="breadcrumbs"
>
  <a
    className="breadcrumbs-item"
    href="/"
    onClick={[Function]}
  >
    <button
      className="breadcrumbs-item-link"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": "auto",
          "lineHeight": "auto",
          "margin": 0,
          "minWidth": "50px",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        Home
      </div>
    </button>
  </a>
  <span
    className="material-icons breadcrumbs-arrow"
    color=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontSize": 24,
        "muiPrepared": true,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
      }
    }
  >
    keyboard_arrow_right
  </span>
  <a
    className="breadcrumbs-item"
    href="/lib"
    onClick={[Function]}
  >
    <button
      className="breadcrumbs-item-link"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": "auto",
          "lineHeight": "auto",
          "margin": 0,
          "minWidth": "50px",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        My Library
      </div>
    </button>
  </a>
  <span
    className="material-icons breadcrumbs-arrow"
    color=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontSize": 24,
        "muiPrepared": true,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
      }
    }
  >
    keyboard_arrow_right
  </span>
  <a
    className="breadcrumbs-item"
    href="/colA"
    onClick={[Function]}
  >
    <button
      className="breadcrumbs-item-link"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": "auto",
          "lineHeight": "auto",
          "margin": 0,
          "minWidth": "50px",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        Collection A
      </div>
    </button>
  </a>
  <span
    className="material-icons breadcrumbs-arrow"
    color=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontSize": 24,
        "muiPrepared": true,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
      }
    }
  >
    keyboard_arrow_right
  </span>
  <span
    className="breadcrumbs-item breadcrumbs-item--active"
  >
    Manual B
  </span>
</div>
`;

exports[`Breadcrumbs single breadcrumb (snapshot) 1`] = `
<div
  className="breadcrumbs"
>
  <span
    className="breadcrumbs-item breadcrumbs-item--active"
  >
    Home
  </span>
</div>
`;
